packages/components/pie-button/src/index.ts

```diff
@@ -196,6 +197,23 @@ export class PieButton extends LitElement implements ButtonProps {
         this._handleClick();
     };
 
+    /**
+     * Template for the loading state
+     *
+     * @private
+     */
+    private renderSpinner (): TemplateResult {
```

Comment:
Do we need tests?

Reply:
Similar to the other comment, we already have visual and component tests in place for this, this PR just refactored the implementation of the logic.